emulators/SyncEmulator.py

```diff
@@ -106,8 +108,10 @@ def print_statistics(self):
 
     def terminated(self, index:int):
         self._progress.acquire()
+        self._done[index] = True
         self._terminated += 1
```

Comment:
I think this `_terminated` variable should die now, the `_done` array implements the same functionality (but in a more robust way)

Reply:
Yup, it's removed in the new commit